File: snippets/csharp/System.Runtime.Versioning/FrameworkName/op_Equality/Operators1.cs
using System;
using System.Runtime.Versioning;

public class Example
{
    public static void Main()
    {
        TestForEquality();
        TestForInequality();
    }

    private static void TestForEquality()
    {
        // <Snippet1>
        FrameworkName version = new(".NET, Version=10.0");
        FrameworkName actualVersion = new($".NET, Version={Environment.Version}");

        Console.WriteLine($"Given Version: {version}");
        Console.WriteLine($"Actual Version: {actualVersion}");
        if (version == actualVersion)
            Console.WriteLine("The versions are the same.");
        else
            Console.WriteLine("The versions are different.");

        Console.WriteLine();

        // The example displays output similar to the following:

        // Given Version: .NET,Version=v10.0
        // Actual Version: .NET,Version=v10.0.10
        // The versions are different.

        // </Snippet1>
    }

    private static void TestForInequality()
    {
        // <Snippet2>
        FrameworkName version = new(".NET, Version=10.0");
        FrameworkName actualVersion = new($".NET, Version={Environment.Version}");

        Console.WriteLine($"Given Version: {version}");
        Console.WriteLine($"Actual Version {actualVersion}");
        if (version != actualVersion)
            Console.WriteLine("The versions are different.");
        else
            Console.WriteLine("The versions are the same.");

        Console.WriteLine();

        // The example displays output similar to the following:

        // Given Version: .NET,Version=v10.0
        // Actual Version: .NET,Version=v10.0.10
        // The versions are different.

        // </Snippet2>
    }
}

File: snippets/visualbasic/System.Runtime.Versioning/FrameworkName/op_Equality/Operators1.vb
' Visual Basic .NET Document
Option Strict On

Imports System.Runtime.Versioning

Module Example
   Public Sub Main()
      TestForEquality()
      TestForInequality()
   End Sub

   Private Sub TestForEquality()
      ' <Snippet1>
      Dim version As New FrameworkName(".NET, Version=10.0")
      Dim actualVersion As New FrameworkName($".NET, Version={Environment.Version}")

      Console.WriteLine($"Given Version: {version}")
      Console.WriteLine($"Actual Version: {actualVersion}")
      If version = actualVersion Then
         Console.WriteLine("The versions are the same.")
      Else
         Console.WriteLine("The versions are different.")
      End If

      Console.WriteLine()

      ' The example displays output similar to the following:

      ' Given Version: .NET,Version=v10.0
      ' Actual Version: .NET,Version=v10.0.10
      ' The versions are different.

      ' </Snippet1>
   End Sub

   Private Sub TestForInequality()
      ' <Snippet2>
      Dim version As New FrameworkName(".NET, Version=10.0")
      Dim actualVersion As New FrameworkName($".NET, Version={Environment.Version}")

      Console.WriteLine($"Given Version: {version}")
      Console.WriteLine($"Actual Version {actualVersion}")
      If version <> actualVersion Then
         Console.WriteLine("The versions are different.")
      Else
         Console.WriteLine("The versions are the same.")
      End If

      Console.WriteLine()

      ' The example displays output similar to the following:

      ' Given Version: .NET,Version=v10.0
      ' Actual Version: .NET,Version=v10.0.10
      ' The versions are different.

      ' </Snippet2>
   End Sub
End Module

